### ISSUE TITLE
Rule proposal: disallow children on DOM element tags when contentEditable

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#713]
React will issue a warning in dev if you use `contentEditable` with children. This can be suppressed with `suppressContentEditableWarning`.

https://github.com/facebook/react/blob/85dcbf83/src/renderers/dom/shared/ReactDOMComponent.js#L187-L193

This should be easy enough to enforce for most cases with a linter rule.

Bad:

```jsx
<div contentEditable>
  Children
</div>
```

Good:

```jsx
<div contentEditable />
```

```jsx
<div contentEditable suppressContentEditableWarning>
  Children
</div>
```

**Comment by lencioni** [comment]
Name: `dom-elements-no-content-editable-with-children`?
